src/Client/GuzzleFileFetcher.php

```diff
@@ -83,39 +83,59 @@ public function fetchMetaData(string $fileName, int $maxBytes): PromiseInterface
 
     /**
      * {@inheritDoc}
+     *
+     * @param array $options
+     *   (optional) Additional request options to pass to the Guzzle client.
+     *   See \GuzzleHttp\RequestOptions.
+     * @param string $url
+     *   (optional) An arbitrary URL from which the target should be downloaded.
+     *   If passed, takes precedence over $fileName.
      */
-    public function fetchTarget(string $fileName, int $maxBytes, array $options = []): PromiseInterface
+    public function fetchTarget(string $fileName, int $maxBytes, array $options = [], string $url = null): PromiseInterface
     {
-        // If $fileName isn't a full URL, treat it as a relative path and prefix
+        // Allow calling code to download the target from an arbitrary URL.
+        $location = $url ?? $fileName;
+        // If $location isn't a full URL, treat it as a relative path and prefix
         // it with $this->targetsPrefix.
         // @todo Revisit the need for this bypass once
         // https://github.com/php-tuf/php-tuf/issues/128 is resolved. This was
         // originally added because of the workaround described there.
-        if (parse_url($fileName, PHP_URL_HOST) === null) {
-            $fileName = $this->targetsPrefix . $fileName;
+        if (parse_url($location, PHP_URL_HOST) === null) {
```

Comment:
why don't we do this only if $url is provided and through an error if 
`parse_url($url, PHP_URL_HOST) === null`
? 
and then only add the prefix if $url is not provided.

Right now you could provide a URL string to $fileName and this would still work.

Reply:
Discussed and I think your insight makes sense. We don't really need the ability to bypass the prefixing, because this PR introduces a way to pass a full URL. So I removed all of this code and simplified it.